pkg/agent/config.go

```diff
@@ -588,8 +590,8 @@ func ValidateAndCombineConfig(log logr.Logger, cfg Config, flags AgentCmdFlags)
 
 	// Validation of --disable-compression.
 	{
-		if flags.DisableCompression && res.AuthMode != VenafiCloudKeypair && res.AuthMode != VenafiCloudVenafiConnection {
-			errs = multierror.Append(errs, fmt.Errorf("--disable-compression can only be used with the %s and %s modes", VenafiCloudKeypair, VenafiCloudVenafiConnection))
+		if !flags.DisableCompression && res.AuthMode != VenafiCloudKeypair && res.AuthMode != VenafiCloudVenafiConnection {
+			errs = multierror.Append(errs, fmt.Errorf("--disable-compression=false can only be used with the %s and %s modes", VenafiCloudKeypair, VenafiCloudVenafiConnection))
 		}
```

Comment:
Since I have no way of knowing if the flag has been passed or not, I propose that we only show this warning when `--disable-compression=false` is passed. It feels a bit off, and another option would have been to remove this validation, but I didn't want to change the code further than I already had.

Reply:
I've replaced this with a log line that will show whenever the user uses `--disable-compression`.